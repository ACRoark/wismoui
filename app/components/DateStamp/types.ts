export type DateStampFormat = 'auto' | 'long' | 'short';
