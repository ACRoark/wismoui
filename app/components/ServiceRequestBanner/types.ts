export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
