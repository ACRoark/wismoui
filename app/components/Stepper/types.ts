import { ReactElement } from 'react';
import { Description } from 'types';

export interface IStepDescription {
  description: Description;
  key: string;
  status: StepperStatus;
  title: ReactElement | string;
}

export enum StepperStatus {
  Active = 'active',
  Completed = 'completed',
  Error = 'error',
  Pending = 'pending',
}
