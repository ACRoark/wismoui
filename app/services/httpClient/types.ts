export interface IHttpError extends Error {
  code: number;
  reason: string;
}

export interface IHttpResponse {
  // tslint:disable-next-line: no-any
  data: any;
  status: number;
  statusText: string;
}

export interface IHttpClient {
  cancel: () => void;
  // tslint:disable-next-line: no-any
  get(url: string, config?: any): Promise<IHttpResponse>;
  // tslint:disable-next-line: no-any
  post(url: string, data: any, config?: any): Promise<IHttpResponse>;
  // tslint:disable-next-line: no-any
  wasCancelled: (error: any) => boolean;
}
