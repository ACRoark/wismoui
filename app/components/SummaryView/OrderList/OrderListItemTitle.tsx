import classnames from 'classnames';
import React from 'react';
import { FormattedMessage } from 'react-intl';

import { IOrderRequestStatusUpdate, OrderRequestType } from 'types';

import messages from './messages';

interface IOrderListItemTitleProps {
  currentOrderRequestStatus: IOrderRequestStatusUpdate | null;
  orderRequestType: OrderRequestType;
}

const OrderListItemTitle: React.FC<IOrderListItemTitleProps> = (
  props: IOrderListItemTitleProps,
): React.ReactElement => {
  const { currentOrderRequestStatus, orderRequestType } = props;

  const hasBPEM = currentOrderRequestStatus && currentOrderRequestStatus.orderRequestStatus === 'NEEDS_VERIFICATION';

  const titleClass = classnames('title', { error: hasBPEM });

  return (
    <div className={titleClass}>
      <FormattedMessage
        {...messages[orderRequestType]}
        values={{
          bpem: '',
          date: '',
          prefix: hasBPEM ? '! ' : '',
        }}
      />
    </div>
  );
};

export default OrderListItemTitle;
