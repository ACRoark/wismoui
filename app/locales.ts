export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
