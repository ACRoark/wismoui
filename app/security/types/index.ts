import IAuthenticationState from 'types/IAuthenticationState';

import IAuthenticationResult from './IAuthenticationResult';

export { IAuthenticationResult };

export default IAuthenticationState;
