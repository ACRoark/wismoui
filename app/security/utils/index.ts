import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
