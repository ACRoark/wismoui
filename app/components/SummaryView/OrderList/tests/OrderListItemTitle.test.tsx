import React from 'react';

import { SUPPORTED_LOCALES } from 'locales';
// TODO: Make this the default type and flip things around
import generate from 'testing/generators/generate';
import { createSnapshotWithIntl, renderWithIntl } from 'testing/utils';
import { IOrderRequestStatusUpdate, OrderRequestStatus, OrderRequestType, OrderRequestTypes } from 'types';

import OrderListItemTitle from '../OrderListItemTitle';

const expectComponentToMatchSnapshot = (
  language: string,
  orderRequestStatus: OrderRequestStatus,
  orderRequestType: OrderRequestType,
): void => {
  const currentOrderRequestStatus: IOrderRequestStatusUpdate = {
    ...generate.orderRequestStatusUpdate(),
    orderRequestStatus,
  };

  const tree = createSnapshotWithIntl(
    <OrderListItemTitle
      currentOrderRequestStatus={currentOrderRequestStatus}
      orderRequestType={orderRequestType}
    />, language);

  expect(tree).toMatchSnapshot();
};

const expectNoConsoleErrors = (
  language: string,
  orderRequestStatus: OrderRequestStatus,
  orderRequestType: OrderRequestType,
): void => {
  const currentOrderRequestStatus: IOrderRequestStatusUpdate = {
    ...generate.orderRequestStatusUpdate(),
    orderRequestStatus,
  };
  const spy = jest.spyOn(global.console, 'error');

  renderWithIntl(
    <OrderListItemTitle
      currentOrderRequestStatus={currentOrderRequestStatus}
      orderRequestType={orderRequestType}
    />, language);

  expect(spy).not.toHaveBeenCalled();
};

describe('<OrderListItemTitle />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    OrderRequestTypes.forEach((requestType: OrderRequestType): void => {
      it(`should not log errors in console when a ${requestType} request with a BPEM is passed (${language})`, (): void =>
        expectNoConsoleErrors(language, 'NEEDS_VERIFICATION', requestType));

      it(`should not log errors in console when a ${requestType} request without a BPEM is passed (${language})`, (): void =>
        expectNoConsoleErrors(language, 'SCHEDULED', requestType));

      it(`should not log errors in console when a ${requestType} request with a BPEM is passed (${language})`, (): void =>
        expectNoConsoleErrors(language, 'NEEDS_VERIFICATION', requestType));

      it(`should not log errors in console when a ${requestType} request without a BPEM is passed (${language})`, (): void =>
        expectNoConsoleErrors(language, 'SCHEDULED', requestType));

      it(`should render correctly when a ${requestType} request with a BPEM is passed (${language})`, (): void =>
        expectComponentToMatchSnapshot(language, 'NEEDS_VERIFICATION', requestType));

      it(`should render correctly when a ${requestType} request without a BPEM is passed (${language})`, (): void =>
        expectComponentToMatchSnapshot(language, 'SCHEDULED', requestType));

      it(`should render correctly when a ${requestType} request with a BPEM is passed (${language})`, (): void =>
        expectComponentToMatchSnapshot(language, 'NEEDS_VERIFICATION', requestType));

      it(`should render correctly when a ${requestType} request without a BPEM is passed (${language})`, (): void =>
        expectComponentToMatchSnapshot(language, 'SCHEDULED', requestType));
    });
  });
});
