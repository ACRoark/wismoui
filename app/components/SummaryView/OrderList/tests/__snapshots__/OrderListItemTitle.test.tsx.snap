// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderListItemTitle /> should render correctly when a MIMO_START request with a BPEM is passed (en) 1`] = `
<div
  className="title error"
>
  ! Start Service 
</div>
`;

exports[`<OrderListItemTitle /> should render correctly when a MIMO_START request with a BPEM is passed (en) 2`] = `
<div
  className="title error"
>
  ! Start Service 
</div>
`;

exports[`<OrderListItemTitle /> should render correctly when a MIMO_START request with a BPEM is passed (es) 1`] = `
<div
  className="title error"
>
  ! Empieza el servicio 
</div>
`;

exports[`<OrderListItemTitle /> should render correctly when a MIMO_START request with a BPEM is passed (es) 2`] = `
<div
  className="title error"
>
  ! Empieza el servicio 
</div>
`;

exports[`<OrderListItemTitle /> should render correctly when a MIMO_START request without a BPEM is passed (en) 1`] = `
<div
  className="title"
>
  Start Service 
</div>
`;

exports[`<OrderListItemTitle /> should render correctly when a MIMO_START request without a BPEM is passed (en) 2`] = `
<div
  className="title"
>
  Start Service 
</div>
`;

exports[`<OrderListItemTitle /> should render correctly when a MIMO_START request without a BPEM is passed (es) 1`] = `
<div
  className="title"
>
  Empieza el servicio 
</div>
`;

exports[`<OrderListItemTitle /> should render correctly when a MIMO_START request without a BPEM is passed (es) 2`] = `
<div
  className="title"
>
  Empieza el servicio 
</div>
`;

exports[`<OrderListItemTitle /> should render correctly when a MIMO_STOP request with a BPEM is passed (en) 1`] = `
<div
  className="title error"
>
  ! Stop Service 
</div>
`;

exports[`<OrderListItemTitle /> should render correctly when a MIMO_STOP request with a BPEM is passed (en) 2`] = `
<div
  className="title error"
>
  ! Stop Service 
</div>
`;

exports[`<OrderListItemTitle /> should render correctly when a MIMO_STOP request with a BPEM is passed (es) 1`] = `
<div
  className="title error"
>
  ! Parada de servicio 
</div>
`;

exports[`<OrderListItemTitle /> should render correctly when a MIMO_STOP request with a BPEM is passed (es) 2`] = `
<div
  className="title error"
>
  ! Parada de servicio 
</div>
`;

exports[`<OrderListItemTitle /> should render correctly when a MIMO_STOP request without a BPEM is passed (en) 1`] = `
<div
  className="title"
>
  Stop Service 
</div>
`;

exports[`<OrderListItemTitle /> should render correctly when a MIMO_STOP request without a BPEM is passed (en) 2`] = `
<div
  className="title"
>
  Stop Service 
</div>
`;

exports[`<OrderListItemTitle /> should render correctly when a MIMO_STOP request without a BPEM is passed (es) 1`] = `
<div
  className="title"
>
  Parada de servicio 
</div>
`;

exports[`<OrderListItemTitle /> should render correctly when a MIMO_STOP request without a BPEM is passed (es) 2`] = `
<div
  className="title"
>
  Parada de servicio 
</div>
`;
